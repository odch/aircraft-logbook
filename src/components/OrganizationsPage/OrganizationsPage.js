import React from 'react'
import Header from '../../containers/HeaderContainer'
import OrganizationsList from '../../containers/OrganizationsListContainer'

const OrganizationsPage = () => (
  <React.Fragment>
    <Header />
    <OrganizationsList />
  </React.Fragment>
)

export default OrganizationsPage
