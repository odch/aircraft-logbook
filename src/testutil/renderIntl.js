import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
