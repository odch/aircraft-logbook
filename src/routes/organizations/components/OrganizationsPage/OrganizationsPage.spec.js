import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import renderIntl from '../../../../testutil/renderIntl'
import OrganizationsPage from './OrganizationsPage'

describe('components', () => {
  describe('OrganizationsPage', () => {
    it('renders correctly', () => {
      const state = {
        firebase: {
          auth: {
            isEmpty: false,
            email: 'test@opendigital.ch'
          },
          profile: {}
        },
        firestore: {},
        main: {
          app: {
            organizations: [{ id: 'org1' }, { id: 'org2' }, { id: 'org3' }]
          }
        },
        organizations: {
          createDialog: {
            open: false,
            data: {
              name: ''
            }
          }
        }
      }

      const store = configureStore()(state)

      const tree = renderIntl(
        <Provider store={store}>
          <Router>
            <OrganizationsPage />
          </Router>
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
