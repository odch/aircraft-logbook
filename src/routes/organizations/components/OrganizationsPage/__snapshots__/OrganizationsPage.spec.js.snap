// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components OrganizationsPage renders correctly 1`] = `
"<header class=\\"MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary\\">
  <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\">
    <h6 class=\\"MuiTypography-root Header-grow-2 MuiTypography-h6 MuiTypography-colorInherit\\"><a class=\\"Header-homeLink-4\\" href=\\"/\\">Open Digital Flugreisebuch</a></h6>
    <div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" data-cy=\\"user-button\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
  </div>
</header>
<div class=\\"MuiGrid-root OrganizationsList-container-99 MuiGrid-container MuiGrid-spacing-xs-3\\" data-cy=\\"organizations-list\\">
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root OrganizationsList-card-100 MuiPaper-rounded\\" data-cy=\\"organization-create-button\\"><button class=\\"MuiButtonBase-root MuiCardActionArea-root OrganizationsList-actionArea-102\\" tabindex=\\"0\\" type=\\"button\\">
        <div class=\\"MuiCardContent-root OrganizationsList-cardContent-101\\"><svg class=\\"MuiSvgIcon-root OrganizationsList-createIcon-103\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\">
            <path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path>
            <path d=\\"M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3v2z\\"></path>
          </svg><span class=\\"MuiTypography-root OrganizationsList-createText-104 MuiTypography-button\\">Neue Organisation erstellen</span></div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
      </button></div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root OrganizationsList-card-100 MuiPaper-rounded\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root OrganizationsList-actionArea-102\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/org1\\">
        <div class=\\"MuiCardContent-root OrganizationsList-cardContent-101\\">
          <h6 class=\\"MuiTypography-root MuiTypography-h6\\">org1</h6>
        </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
      </a></div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root OrganizationsList-card-100 MuiPaper-rounded\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root OrganizationsList-actionArea-102\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/org2\\">
        <div class=\\"MuiCardContent-root OrganizationsList-cardContent-101\\">
          <h6 class=\\"MuiTypography-root MuiTypography-h6\\">org2</h6>
        </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
      </a></div>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root OrganizationsList-card-100 MuiPaper-rounded\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root OrganizationsList-actionArea-102\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/org3\\">
        <div class=\\"MuiCardContent-root OrganizationsList-cardContent-101\\">
          <h6 class=\\"MuiTypography-root MuiTypography-h6\\">org3</h6>
        </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
      </a></div>
  </div>
</div>"
`;
